### ISSUE TITLE
Old vocabularies may be fetched during the check

### ISSUE THREAD

**Issue by berezovskyi** [oslc-op/oslc-specs#40]
There's a fundamental problem here in the way the tables are built for the cross-check. Instantiation of VocabularyCheck loads a vocabulary document into memory (good), and runs the checks on that vocabulary, which in turn loads many of the resources referenced from vocabulary (bad). It is bad because those references might be part of a second or subsequent vocabulary document to be checked, but which has not yet been loaded or processed. So references between vocabularies, or references from shapes to items loaded from resources that were referenced from vocabularies can end up looking at out-of-date vocabularies loaded from their real locations on the internet, and not from the source being checked. **The issue can also arise between shapes that reference other shapes (via the allowedValues properties, etc.).** 

To fix this, separate vocabulary loading from checking, and pre-load all the vocabs first (not only the vocabs requested on the command line, but all the shapes on the command line as well). Meanwhile, work around this by setting @base in the source to make the URIs for terms reflect the intended target, and use the -x command line option to exclude that (presumably older) target. 

_From @ndjc_

**Comment by ndjc** [comment]
Right. I've not tried the equivalent work-around for inter-shape references, so I don't know that problem can be avoided right now.
